src/dex/uniswap-v3/contract-math/Oracle.ts

```diff
@@ -93,12 +93,12 @@ export class Oracle {
     let beforeOrAt;
     let atOrAfter;
     while (true) {
-      i = (l + r) / 2;
+      i = Math.floor((l + r) / 2);
 
       beforeOrAt = state.observations[i % cardinality];
 
       // we've landed on an uninitialized tick, keep searching higher (more recently)
-      if (!beforeOrAt.initialized) {
+      if (!beforeOrAt || !beforeOrAt.initialized) {
```

Comment:
In Solidity, accessing a member of an undefined struct won't cause any issues since the values will be returned as 0 or empty. But in Javascript, accessing a memory on a undefined object will cause the system to crash. We simply added a check that the object should be defined before accessing its member here: https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/Oracle.sol#L169